fabric-message-api-v1/src/main/java/net/fabricmc/fabric/api/message/v1/ServerChatEvents.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.message.v1;
+
+import net.minecraft.network.MessageType;
+import net.minecraft.network.encryption.SignedChatMessage;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.filter.Message;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import net.minecraft.util.registry.RegistryKey;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public final class ServerChatEvents {
+	private ServerChatEvents() {
+	}
+
+	/**
+	 * An event that is called when a player sends a chat message to all players, either via the
+	 * chat hud or the {@code /say} command. Mods can use this event to block the message.
+	 */
+	public static final Event<AllowChatMessage> ALLOW_CHAT_MESSAGE = EventFactory.createArrayBacked(AllowChatMessage.class, handlers -> (message, sender, typeKey) -> {
```

Comment:
It is unclear when this fires, does it act on the inbound side from player packets or just before broadcasting such that it includes mod generated/replayed message? (some concern applies to other events)

Reply:
It is fired just before sending, so this does include the mod-generated message. @sfPlayer1 Should this be renamed or is a javadoc fine?

Reply:
javadoc may be fine for this, but the "send" terminology is a especially misleading for the events below - the client sends and the server sends.. could use "broadcast" instead?